### PR TITLE
News Downloader: Select Custom download folder

### PR DESCRIPTION
**Pull request by mwoz123**
Ability to select custom download folder

### CODE REVIEW

File: plugins/newsdownloader.koplugin/main.lua [hunk 1]
```diff
@@ -10,40 +10,33 @@ local logger = require("logger")
...
-    if not ok then return end
-    return xmlhandler.root
-end
 
 function NewsDownloader:init()
     self.ui.menu:registerToMainMenu(self)
 end
 
 function NewsDownloader:addToMainMenu(menu_items)
+
```
Comment (Frenzie) [inline plugins/newsdownloader.koplugin/main.lua]: stray newline

File: plugins/newsdownloader.koplugin/main.lua [hunk 2]
```diff
@@ -10,40 +10,33 @@ local logger = require("logger")
...
     if not initialized then
-        news_download_dir_path = ("%s/%s/"):format(DataStorage:getDataDir(), news_download_dir_name)
+        local news_downloader_settings = LuaSettings:open(DataStorage:getSettingsDir().."/" .. news_downloader_config_file)
+        if news_downloader_settings:has(config_key_custom_dl_dir) then
+            news_download_dir_path = news_downloader_settings:readSetting(config_key_custom_dl_dir)
+	else
```
Comment (Frenzie) [inline plugins/newsdownloader.koplugin/main.lua]: 4 spaces, not tabs

File: plugins/newsdownloader.koplugin/main.lua [hunk 3]
```diff
@@ -10,40 +10,33 @@ local logger = require("logger")
...
 
 function NewsDownloader:init()
     self.ui.menu:registerToMainMenu(self)
 end
 
 function NewsDownloader:addToMainMenu(menu_items)
+
     if not initialized then
-        news_download_dir_path = ("%s/%s/"):format(DataStorage:getDataDir(), news_download_dir_name)
+        local news_downloader_settings = LuaSettings:open(DataStorage:getSettingsDir().."/" .. news_downloader_config_file)
```
Comment (houqp) [inline plugins/newsdownloader.koplugin/main.lua]: Why not use `format` everywhere to keep it consistent?

**Comment by houqp** [comment]
Nice work :+1:

**Pull request merged**